### PR TITLE
FIX: tck as sft for header check

### PR DESCRIPTION
**Pull request by AntoineTheb**
Address #379.

`dipy.io.utils.is_header_compatible` does not handle `.tck`s. As such I went through scripts to make sure none passed `.tck` filenames to this function and loaded them as `StatefulTractogram` if they could have.

@frheault @arnaudbore @GuillaumeTh 

~Tests are still running, will update once it's done~ All tests have passed

### CODE REVIEW

File: scripts/scil_label_and_distance_maps.py [hunk 1]
```diff
@@ -51,11 +51,11 @@ def main():
     assert_outputs_exist(parser, args, [args.output_label,
                                         args.output_distance])
 
-    is_header_compatible(args.in_bundle, args.in_centroid)
-
     sft_bundle = load_tractogram_with_reference(parser, args,
                                                 args.in_bundle)
 
+    is_header_compatible(sft_bundle, args.in_centroid)
```
Comment (arnaudbore) [inline scripts/scil_label_and_distance_maps.py:57]: here args.in_centroid can also be a tck right ?

**Review by arnaudbore — APPROVED** [review]
One last fix but otherwise LGTM.

File: scripts/scil_label_and_distance_maps.py (hunk 1 above)
Comment (AntoineTheb) [reply scripts/scil_label_and_distance_maps.py]: Indeed ! Fixed in 32dfe8f

**Review by arnaudbore — APPROVED** [review]
LGTM

**Comment by arnaudbore** [comment]
Quick discussion with Francois.

We should have a custom: check_compatibility_tractogram_with_reference()
We should check the extension before going for a sft.

**Comment by AntoineTheb** [comment]
Well if the idea is to have a cleaner solution, I thought about writing a wrapper around dipy's `is_header_compatible` to make sure that only tcks would be loaded for a header check, and that they would be loaded with their reference. While it seems like a simple enough idea, you may notice that there are many more references to `is_header_compatible` in scilpy than what was modified here. This is because most scripts didn't need this added check, and so the added function would only save a few lines at best. So it would require a lot of work, potentially breaking many things, without too much added value.

If the idea was to save computation, you may notice that most scripts would have loaded the tractograms right after the check anyways, so it wouldn't save that much ressources.

All in all, I agree it would be cleaner to have a function like you're proposing, but I think it is out of scope for this PR (which let's remember addresses a bug that a user encountered).

**Review by arnaudbore — APPROVED** [review]

**Pull request merged**